taojieTan/assignments/fastSlow/lc141/lc141.py

```diff
@@ -0,0 +1,80 @@
+# Problem Statement #
+
+# Given the head of a Singly LinkedList, write a function to determine if the LinkedList has a cycle in it or not.
+
+class Node:
+  def __init__(self, value, next=None):
+    self.value = value
+    self.next = next
+
+
+def has_cycle(head):
+  # TODO: Write your code here
+
+# input: a singly linkedlist with integer values
+# output: boolean value 
+# edge cases: 1. empty array => return False
+#             2. 1 node => can be True if it points to itself; otherwise, False
+
+# we can apply slow and fast pointers pattern for this question
+# we set fast pointer two steps faster than slow pointer, so that if there is a cycle in the linkedlist, 
+# fast pointer will eventually catch the slow pointer.
```

Comment:
It may be helpful to go into a bit more detail for this approach, more like a step-by-step of what your code will do in high-level terms.